reindexer/reindex_request_processor/src/test/scala/uk/ac/wellcome/platform/reindex/processor/services/ReindexWorkerServiceTest.scala

```diff
@@ -0,0 +1,184 @@
+                vhsBucket,
+                vhsTable,
+                ReindexableRecord(id, originalVersion))
+
+              sendMessage(
+                queue,
+                reindexRequest.copy(desiredVersion = updatedVersion))
+
+              eventually {
+                assertStored[ReindexableRecord](
+                  vhsBucket,
+                  vhsTable,
+                  ReindexableRecord(id, updatedVersion))
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  it("an existing record is not updated by an older version") {
+    withLocalSqsQueue { queue =>
+      withLocalS3Bucket { vhsBucket =>
+        withLocalDynamoDbTable { vhsTable =>
+          withReindexWorkerService(vhsTable, vhsBucket, queue) { _ =>
+            val originalVersion = 2
+            val olderVersion = 1
+
+            val reindexRequest = ReindexRequest(id, originalVersion)
+            sendMessage(queue, reindexRequest)
+
+            eventually {
+              assertStored[ReindexableRecord](
+                vhsBucket,
+                vhsTable,
+                ReindexableRecord(id, originalVersion))
+
+              sendMessage(
+                queue,
+                reindexRequest.copy(desiredVersion = olderVersion))
+
+              assertQueueEmpty(queue)
+              eventually {
+                assertStored[ReindexableRecord](
+                  vhsBucket,
+                  vhsTable,
+                  ReindexableRecord(id, originalVersion))
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+
+  it("fails if saving to S3 fails") {
+    withLocalSqsQueueAndDlq {
+      case QueuePair(queue, dlq) =>
+        withLocalDynamoDbTable { vhsTable =>
+          withReindexWorkerService(vhsTable, Bucket("wrongBucket"), queue) {
+            _ =>
+              sendMessage(queue, ReindexRequest(id = id, desiredVersion = 1))
+
+              eventually {
+                assertQueueEmpty(queue)
+                assertQueueHasSize(dlq, 1)
+              }
+          }
+        }
+    }
+  }
+
+  private def sendMessage(queue: Queue, reindexRequest: ReindexRequest) = {
+    sqsClient.sendMessage(
+      queue.url,
+      toJson(
+        NotificationMessage(
+          "snsID",
+          "snsTopic",
+          "snsSubject",
+          toJson(reindexRequest).get)).get)
+  }
+
+  private def withReindexWorkerService[R](
+    vhsTable: Table,
+    vhsBucket: Bucket,
+    queue: Queue)(testWith: TestWith[ReindexWorkerService, R]) = {
+    withActorSystem { actorSystem =>
+      withMetricsSender(actorSystem) { metricsSender =>
+        withTypeVHS[ReindexableRecord, EmptyMetadata, R](
+          bucket = vhsBucket,
+          table = vhsTable) { versionedHybridStore =>
+          val sqsStream = new SQSStream[NotificationMessage](
+            actorSystem,
+            asyncSqsClient,
+            SQSConfig(queue.url, waitTime = 1 second, maxMessages = 1),
+            metricsSender)
```

Comment:
minor: you could also use `withSQSStream[NotificationMessage, R](actorSystem, queue, metricsSender)` to make this simpler